### PR TITLE
feat(events): new events module with hosting plus functions codebases (#94)

### LINKED ISSUE #94: Community event management solution
- ability to add an "event"
- listing of events
- viewing an event

### PR DESCRIPTION
**Pull request by andrew-cunliffe**
closes #94

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,16 +2,10 @@
...
 # project local files
 *.local.ts
-
-# compiled output
-/dist
-/tmp
-/out-tsc
-# Only exists if Bazel was run
-/bazel-out
+*.local.txt
```
Comment (eddiejaoude) [inline .gitignore]: May be use `*.local.*`?

File: functions/src/google/places.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { GooglePlaceFind, GooglePlaceSearch } from '../models';
+import { CallableContext } from 'firebase-functions/lib/providers/https';
+
+export const GooglePlacesSearch = functions.https.onCall(async (data: any, context: CallableContext): Promise<any> => {
+    const uid = data.sessionId ? data.sessionId : v4();
+    const value = data.placeName;
+    const type = data.type;
+
+    const method = 'POST';
+    const uri = `https://maps.googleapis.com/maps/api/place/autocomplete/json`;
```
Comment (eddiejaoude) [inline functions/src/google/places.ts]: This could just be single quotes

File: functions/src/google/test.json [hunk 3]
```diff
@@ -0,0 +1,228 @@
+{
```
Comment (eddiejaoude) [inline functions/src/google/test.json]: Does this get used? I could not find `test.json` in the PR

File: hosting/src/app/account/account.module.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { NgModule } from '@angular/core';
 
-import { DesignModule } from '@cm/design/design.module';
+import { DesignModule } from '../../design/design.module';
```
Comment (eddiejaoude) [inline hosting/src/app/account/account.module.ts]: Not using `@cm` anymore?

File: hosting/src/app/account/profile/profile.component.html [hunk 5]
```diff
@@ -10,6 +10,8 @@ <h2>{{ account.alias }} </h2>
     </div>
 </section>
 
+<pre>{{ account | json }}</pre>
```
Comment (eddiejaoude) [inline hosting/src/app/account/profile/profile.component.html]: Do we want to keep the `pre` in still?

File: hosting/src/core/models/event.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import * as firebase from 'firebase';
+
+import { GooglePlace } from './google-place-find';
+
+export interface Event {
+    uid: string;
+    type: 'stream' | 'meetup' | 'conference' | 'hackathon' | 'workshop';
```
Comment (eddiejaoude) [inline hosting/src/core/models/event.ts]: Should probably move this to a type to be used in multiple places

File: .gitignore (hunk 1 above)
Comment (andrew-cunliffe) [reply .gitignore]: you can't have a `*` for extensions on gitignore

File: functions/src/google/places.ts (hunk 2 above)
Comment (andrew-cunliffe) [reply functions/src/google/places.ts]: ah yea since things have changed here I can fix this

File: functions/src/google/test.json (hunk 3 above)
Comment (andrew-cunliffe) [reply functions/src/google/test.json]: nope this was something I noticed too I have some clean up :)

File: hosting/src/app/account/account.module.ts (hunk 4 above)
Comment (andrew-cunliffe) [reply hosting/src/app/account/account.module.ts]: good spot :)

File: hosting/src/app/account/profile/profile.component.html (hunk 5 above)
Comment (andrew-cunliffe) [reply hosting/src/app/account/profile/profile.component.html]: :) yup need to clean this up

File: hosting/src/core/models/event.ts (hunk 6 above)
Comment (andrew-cunliffe) [reply hosting/src/core/models/event.ts]: ill let you clean up the model sometime ;)

Comment (eddiejaoude) [reply hosting/src/core/models/event.ts]: Sure 👍

**Pull request merged**